code/API_definitions/EdgeCloud_LcM.yaml

```diff
@@ -0,0 +1,658 @@
+openapi: 3.0.3
+info:
+  title: Edge Cloud API
+  description:  |
+ 
+    Edge Cloud API for artifact managment, application management and resource management.  
+    # Introduction
+    APIs defined in this version of the specification can be categorized into the following areas:
+    * __Application Management__ - Deploy and remove applications within artifacts
+    * __Application Artifacts Management__ - Management of application descriptors, binaries, charts and packages 
+    * __MEC Platform information__ - Retrieves all the Edge nodes available according to some defined parameters
+    # Relevant terms and definitions
+    This section provides definitions of terminologies commonly referred to throughout the API descriptions.
+ 
+    * __Application Provider__ - An application developer, onboarding his/her edge application on operator platform (MEC).    
+    * __Artifact__ - Descriptor, binary image, charts or any other package assosiated with the application.
+    * __OP__ - Operator Platform.
+    * __Region__ - Human readable name of the geografical zone of the MEC. Defined by the OP.
+    # API Functionality
+    __Application Management__
+    * __onboardApplication__ - Submits an application details to an OP. Based on the details provided,  OP shall do bookkeeping, resource validation and other pre-deployment operations. 
+    * __removeApplication__ - Removes an application from an OP
+ 
+    __Application Artifacts Management__
+    * __onboardArtifact__  Submits an artifact to operator platform
```

Comment:
Should we define what is an artifact for clarity and its relation with Application?

Reply:
I agree, for us an Artifact is the actual binary data, i.e. POSTing an artifact is in fact uploading a binary file to the platform, not storing a reference.

Reply:
In the v0.2.0 we defined this entities:

- **Application** - The Application Provider submit application metadata to the Operator Platform.

- **Application Instance** - An Operator Platform instantiate an Application on an Edge Cloud Node when the Application Provider resquest it.

And as discussed, artifact is no longer used